### PR TITLE
Add optional support for hidden files/directories

### LINKED ISSUE #1808: Optionally support hidden files/directories in actix-files
## Expected Behavior
It should be possible to serve static hidden files/directory.

## Current Behavior
A leading dot in a url fragment is currently not allowed, even if a file or directory with the exact name exists.

## Steps to Reproduce (for bugs)
Instantiate a server with a `Files` service to serve static files, then try to access an existing hidden file or directory.

## Context
This is particularly useful when trying to use certbot to perform its challenge. Probably its not good to allow this by default, I would like the feature to be enabled using a boolean flag.

## Your Environment
* rustc 1.48.0 (7eac88abb 2020-11-16)
* actix-files: 0.4.1
* actix-web: 3.3.0

### PR DESCRIPTION
**Pull request by dodomorandi**
## PR Type

Feature

## PR Checklist
Check your PR fulfills the following:

- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt

## Overview
Introduce the possibility of serving hidden files and directory, explicitly allowing a leading dot in the url segments.

Closes #1808

### CODE REVIEW

File: actix-files/src/files.rs [hunk 1]
```diff
@@ -213,6 +216,13 @@ impl Files {
 
         self
     }
+
+    /// Enables serving hidden files and directories, allowing a trailing dots in url fragments.
```
Comment (robjtede) [inline actix-files/src/files.rs]:
```suggestion
    /// Enables serving hidden files and directories, allowing a leading dots in url fragments.
```

Comment (dodomorandi) [reply actix-files/src/files.rs]: :facepalm: Sorry, I literally wrote `trailing` everywhere :roll_eyes:

File: actix-files/CHANGES.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Changes
 
 ## Unreleased - 2020-xx-xx
+* Optionally support hidden files/directories.
```
Comment (robjtede) [inline actix-files/CHANGES.md]:
```suggestion
* Optionally support hidden files/directories. [#1811]

[#1811]: https://github.com/actix/actix-web/pull/1811
```

File: actix-files/src/path_buf.rs [hunk 3]
```diff
@@ -15,12 +15,19 @@ impl FromStr for PathBufWrap {
...
+impl PathBufWrap {
+    /// Parse a path, giving the choice of allowing hidden files to be considered valid segments.
+    pub fn parse_path(path: &str, hidden_files: bool) -> Result<Self, UriSegmentError> {
         let mut buf = PathBuf::new();
 
         for segment in path.split('/') {
             if segment == ".." {
                 buf.pop();
-            } else if segment.starts_with('.') {
+            } else if segment.starts_with('.') && !hidden_files {
```
Comment (robjtede) [inline actix-files/src/path_buf.rs]: do the cheaper op first

```suggestion
            } else if !hidden_files && segment.starts_with('.') {
```

**Comment by dodomorandi** [comment]
@robjtede Thank you for your review and your patience. Should I do anything about the codecov? Most of the coverage issues shown do not seem to be very meaningful...

**Comment by robjtede** [comment]
> Should I do anything about the codecov?

It's fine, no action required on your part.

**Review by otavio — APPROVED** [review]

**Pull request merged**